### PR TITLE
Fix race in apiserver_test

### PR DESCRIPTION
**Pull request by gaurav1086**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
/kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
When a go loop launches goroutines, it is a common mistake to assume
that the routines can reference the loop value from their specific loop
iteration. Actually, all goroutines share the same reference to the loop
veriable, and they typically all see the last value. To correct this, the loop variable must be captured as a parameter to the anonymous func.

**Which issue(s) this PR fixes**:

Fixes #
N/A
**Special notes for your reviewer**:
N/A
**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/endpoints/apiserver_test.go [hunk 1]
```diff
@@ -2104,7 +2104,7 @@ func TestWatchTable(t *testing.T) {
 
 	s := metainternalversionscheme.Codecs.SupportedMediaTypes()[0].Serializer
 
-	tests := []struct {
+	type _test_ struct {
```
Comment (neolit123) [inline staging/src/k8s.io/apiserver/pkg/endpoints/apiserver_test.go:2107]: local names should use camelCase (no underscores)

**Comment by neolit123** [comment]
running tests so that you can debug.
/ok-to-test

please fix https://github.com/kubernetes/kubernetes/pull/90490#discussion_r415136859

**Comment by gaurav1086** [comment]
> running tests so that you can debug.
> /ok-to-test
> 
> please fix [#90490 (comment)](https://github.com/kubernetes/kubernetes/pull/90490#discussion_r415136859)

Changes done.

**Pull request closed without merging**